packages/marshaller/src/ddl2/activities/filter.ts

```diff
@@ -124,20 +136,30 @@ ColumnMapping.prototype._class += " ColumnMapping";
 export class Filter extends PropertyExt {
     private _owner: Filters;
 
-    @publish(null, "set", "Activity", function (this: Filter) { return this.visualizationIDs(); }, { optional: true })
+    @publish(null, "set", "Activity", function (this: Filter) { return this.visualizationIDs(); }, {
+        optional: true,
+        validate: (w: Filter): boolean => w.visualizationIDs().indexOf(w.source()) >= 0
+    })
     source: publish<this, string>;
     @publish([], "propertyArray", "Mappings", null, { autoExpand: ColumnMapping })
     mappings: publish<this, ColumnMapping[]>;
 
     validate(prefix: string): IActivityError[] {
         let retVal: IActivityError[] = [];
-        if (this.visualizationIDs().indexOf(this.source()) < 0) {
+        if (!this["source_valid"]()) {
```

Comment:
Why not `!this.source_valid()` like the other `_valid` conditionals?

Reply:
It should - I will fix